File: bindings/python/rpc/test/FirteXRPCClientTest.py
import sys
sys.path.append('../')
sys.path.append('../../')
from firtex import ThriftSearchService
from firtex.ttypes import *
from rpc import FirteXRPCClient

print 'test syntaxSearch'
client = FirteXRPCClient.FirteXRPCClient('localhost', 19801)
result = client.syntaxSearch('query=BODY:2;field_select=BODY:snippet')
print result

print 'test search'
fields = ['BODY:snippet']
cmd = QueryCommand('BODY:2', None, None, 0, 10, fields)
result = client.search(cmd)
print result

File: bindings/php/rpc/test/FirteXRPCClientTest.php
<?php
include_once dirname(__FILE__).'/../FirteXRPCClient.php';

echo '===test syntaxSearch';
$client = new FirteXRPCClient('localhost', '19801');
$result = $client->syntaxSearch('query=BODY:2;field_select=BODY:snippet');
print_r($result);

echo '===test search';
$client = new FirteXRPCClient('localhost', '19801');
$vals = array('query' => 'BODY:2',
              'fieldSelect' => array(0 => 'BODY:snippet'));
$cmd = new firtex_QueryCommand($vals);
$result = $client->search($cmd);
print_r($result);
?>